internal/envelope/conn/envelope_conn.go

```diff
@@ -355,3 +354,37 @@ func (e *EnvelopeConn) rpc(request *protos.EnvelopeMsg) (*protos.WeaveletMsg, er
 	}
 	return msg, nil
 }
+
+// verifyWeaveletInfo verifies the information sent by the weavelet.
+func verifyWeaveletInfo(wlet *protos.WeaveletInfo) error {
+	if wlet == nil {
+		return fmt.Errorf(
+			"the first message from the weavelet must contain weavelet info")
+	}
+	if wlet.DialAddr == "" {
+		return fmt.Errorf("empty dial address for the weavelet")
+	}
+	if err := checkVersion(wlet.Version); err != nil {
+		return err
+	}
+	return nil
+}
+
+// checkVersion checks that the deployer API version the deployer was built
+// with is compatible with the deployer API version the app was built with,
+// erroring out if they are not compatible.
+func checkVersion(appVersion *protos.SemVer) error {
```

Comment:
Doesn't this overlap with the changes Michael is making? Should we be not mucking with version checking in this change? Or is this just code movement that will go away soon?